Nautilus/Documentation/tutorials/vehicle-module.md

```diff
@@ -0,0 +1,137 @@
+Basically, all the modules are stored locally by Nautilus in order to run actions, set the modules settings in game, such as the crush depth if provided, the cooldown if provided, etc...
+
+# Create a module with the gadget
+
+Here are some examples of uses of the gadget, used in many ways.  
+You are not limited, you can do everything you want with those.
+
+## Passive module setting a new crush depth
+
+Let's create a custom module setting the depth to, let's say, 1700m for the SeaMoth.  
+So, first, let's make our prefab info.
+
+```csharp
+var prefabInfo = PrefabInfo.WithTechType("SeamothDepthUpgrade", "Seamoth Depth Module MK.4", "Dive down to 1700 meters!!! Let's meet the Sea Dragon!")
+    .WithIcon(SpriteManager.Get(TechType.HullReinforcementModule3));
+```
+
+Then, we're making the Custom Prefab based on Reinforced Hull prefab.
+
+```csharp
+var prefab = new CloneTemplate(prefabInfo, TechType.HullReinforcementModule3);
+```
+
+Now, let's quickly set the basic settings.
+
+```csharp
+prefab.SetRecipe(new Crafting.RecipeData()
+{
+    craftAmount = 1,
+    Ingredients = new List<CraftData.Ingredient>()
+    {
+        new CraftData.Ingredient(TechType.HullReinforcementModule3),
+        new CraftData.Ingredient(TechType.AluminumOxyde, 2),
+        new CraftData.Ingredient(TechType.AdvancedWiringKit, 3)
+    }
+})
+    .WithFabricatorType(CraftTree.Type.SeamothUpgrades)
+    .WithStepsToFabricator("SeamothModules")
+    .WithCraftingTime(5f);
+```
+
+And finally, let's make the part that interest us: Adding the Upgrade Module Gadget.
+
+```csharp
+// This first function defines the equipment type and the quick slot type.
+prefab.SetUpgradeModule(EquipmentType.SeamothModule, QuickSlotType.Passive)
+    .WithDepthUpgrade(1700f, true)  // 1700 is the depth in meters. The boolean after defines if we want the value to be absolute or "relative", added to the default depth. 
+                                    // If it is on false, it will set the new crush depth to 1900 meters because the default depth of the Seamoth is 200 meters.
+                                    // Otherwise, it will set the new crush depth on 1700 meters.
+    .WithOnModuleAdded((Vehicle vehicleInstance, int slotId) =>
+    {
+        Subtitles.Add("Well done! The module is working! The new depth is 1700 meters.");
+    })
+    .WithOnModuleRemoved((Vehicle vehicleInstance, int slotId) => {
+        Subtitles.Add("Ah... You removed the upgrade. Take care of your hull.");
+    });
+```
+> [!NOTE]
+> You can do basically everything you want in these delegates, such as trigger story events, destroy the vehicle, add components to the vehicle instance, etc...
+
+## Chargeable self-destruct module
+
+Admitting we've already done the prefab info and the custom prefab, let's directly work only with the upgrade module gadget.
```

Comment:
Odd choice of "admitting" here. It makes sense, but at the same time it really doesn't...

Reply:
What should I say ?

Reply:
I'm struggling with figuring out how to word it ngl. "Since"?

Reply:
Assuming ?

Reply:
Well there is no assumption happening here, since the code examples above already mention PrefabInfo.

Reply:
in fact, it's like if there was a reset between the two examples, they are independent, i just tell to assume that we've already done a prefab info and custom prefab for this new example I guess